bgpd/bgp_route.c

```diff
@@ -2492,8 +2492,11 @@ bool subgroup_announce_check(struct bgp_dest *dest, struct bgp_path_info *pi,
 		    || (!reflect && !transparent
 			&& IN6_IS_ADDR_LINKLOCAL(&peer->nexthop.v6_local)
 			&& peer->shared_network
-			&& (from == bgp->peer_self
-			    || peer->sort == BGP_PEER_EBGP))) {
+			&& ((from == bgp->peer_self
+				&& peer->sort == BGP_PEER_EBGP)
+				|| (from != bgp->peer_self
+					&& IN6_IS_ADDR_LINKLOCAL(&attr->mp_nexthop_local)
```

Comment:
Can you put a reference to RFC where it clearly explains this behavior?

Reply:
added to the PR description